integration-tests/contract-security-ticket-callback-lambdas.spec.ts

```diff
@@ -0,0 +1,263 @@
+                    PUSH nat 1;
+                    PUSH string "test";
+                    DIG 2;
+                    PAIR 3;
+                  } ;
+                DROP; # DROP storage and input
+                DROP; # DROP lambda
+                UNIT;
+                NIL operation;
+                PAIR;
+              };}`,
+          init: 'Unit'
+        });
+
+        await opGetter.confirmation();
+        expect(opGetter.hash).toBeDefined();
+        expect(opGetter.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+
+      } catch (error: any) {
+        expect(error.message).toContain('michelson_v1.inconsistent_types');
+      }
+      done();
+    });
+
+    it("Verify contract for ticket is not created with a lambda - string", async (done) => {
+      try {
+        const opGetter = await Tezos.contract.originate({
+          code: ` { parameter string;
+            storage unit;
+            code
+              {
+                # Trying to use a lambda in order to create a "ticket"
+                LAMBDA
+                  string
+                  (option (ticket string))
+                  {
+                    PUSH nat 1;
+                    PUSH string "test";
+                    DIG 2;
+                    PAIR 3;
+                    SOME;
+                  } ;
+                DROP; # DROP storage and input
+                DROP; # DROP lambda
+                UNIT;
+                NIL operation;
+                PAIR;
+              };}`,
+          init: 'Unit'
+        });
+
+        await opGetter.confirmation();
+        expect(opGetter.hash).toBeDefined();
+        expect(opGetter.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+
+      } catch (error: any) {
+        expect(error.message).toContain('michelson_v1.inconsistent_type_sizes');
+      }
+      done();
+    });
+
+    it("Verify contract for ticket is not created - fail with no mutez", async (done) => {
+      try {
+        const opRogue = await Tezos.contract.originate({
+          code: ` { parameter address;
+            storage (option (ticket string));
+            code
+              {
+                UNPAIR; 
+                CONTRACT (ticket string);
+                IF_NONE
+                  {
+                     PUSH string "error 1";
+                     FAILWITH;
+                  }
+                  {};
+                PUSH mutez 0;
+                DIG 2;
+                IF_NONE
+                  {
+                     PUSH string "error 2";
+                     FAILWITH;
+                  }
+                  {};
+                TRANSFER_TOKENS; 
+                NIL operation;
+                SWAP;
+                CONS;
+                PUSH nat 1;
+                PUSH string "test";
+                TICKET; 
+                SOME;
+                SWAP;
+                PAIR;
+              };}`,
+          init: '"Some (Pair \"KT1Pmz71yHTCmDHiJ9quFpb9cKANeeb2evV2\" \"test\" 2)"'
```

Comment:
Is it allowed to pass a ticket value like that? I thought they were only created on chain.